### PR TITLE
chat message manager: don't send messages if player is on tutorial island

### PR DESCRIPTION
**Pull request by Trevor159**
If we send messages on tutorial island it looks like this and can get in the way of the tutorial.

### CODE REVIEW

**Comment by Trevor159** [comment]
updated to use region ids

File: runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java [hunk 1]
```diff
@@ -56,6 +58,8 @@
 @Singleton
 public class ChatMessageManager
 {
+	private static final Set<Integer> tutIslandRegions = ImmutableSet.of(12336, 12335, 12592, 12080, 12079, 12436);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java]: UPPER_CASE

File: runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java [hunk 2]
```diff
@@ -550,6 +554,12 @@ public void process()
 
 	private void add(QueuedMessage message)
 	{
+		// Do not send message if the player is on tutorial island
+		if (TUTORIAL_ISLAND_REGIONS.contains(client.getLocalPlayer().getWorldLocation().getRegionID()))
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java]: You should also null-check player, probably. As add can be called from login screen, I think, even though it does nothing.

File: runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java [hunk 3]
```diff
@@ -550,6 +555,13 @@ public void process()
 
 	private void add(QueuedMessage message)
 	{
+		// Do not send message if the player is on tutorial island
+		final Player player = client.getLocalPlayer();
+		if (player == null || TUTORIAL_ISLAND_REGIONS.contains(player.getWorldLocation().getRegionID()))
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java]: This should be changed to player != null && to not change existing behaviour

**Pull request merged**